### PR TITLE
Add exchangeInfo to redux

### PR DESCRIPTION
**Pull request by peachbits**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203666515493951

### CODE REVIEW

File: src/components/scenes/GuiPluginListScene.tsx [hunk 1]
```diff
@@ -341,6 +343,7 @@ class GuiPluginList extends React.PureComponent<Props, State> {
     // Filter disabled plugins:
     const activePlugins = bestOfPlugins(accountPlugins, accountReferral, undefined)
     plugins = plugins.filter(plugin => !activePlugins.disabled[plugin.pluginId])
+    plugins = plugins.filter(plugin => !disablePlugins[plugin.pluginId])
```
Comment (paullinator) [inline src/components/scenes/GuiPluginListScene.tsx:346]: Change to

`plugins = plugins.filter(plugin => !disablePlugins[plugin.id])`

`pluginId` is not specific enough as it's shared between different payment methods and also with plugins in the FiatPlugins framework.

File: src/plugins/gui/fiatPluginTypes.ts [hunk 2]
```diff
@@ -59,6 +60,7 @@ export interface FiatPluginFactoryArgs {
   // io: {
   //   log: EdgeLog, // scoped logs
   // }
+  disablePlugins: DisablePluginMap
```
Comment (paullinator) [inline src/plugins/gui/fiatPluginTypes.ts:63]: Per convo with @DunningKrugR , looks like we need more granular ability to disable provider plugins and not reuse the buy/sell plugin list.

I would recommend a structure like the following

```
{
    exchangeInfo: {
        buy:  {
            disablePlugins:  {
                moonpay: true, // This refers to a high level plugin.id
                wyre: true,
                creditcard: { // Applies to the creditcard FiatPlugin
                    moonpay: true // Within the creditcard FiatPlugin, this applies to the provider id
                }
            },
        },
        sell: {
            disablePlugins:  {
                simplex: true
                iach: {
                    moonpay: true
                }
            },
        }
    }
}
```

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/actions/CryptoExchangeActions.tsx [hunk 3]
```diff
@@ -140,11 +146,16 @@ async function fetchSwapQuote(state: RootState, request: EdgeSwapRequest): Promi
     console.log(`Preferring ${preferPluginId} from ${reason}`)
   }
 
+  const disablePluginsMap: { [pluginId: string]: true } = {}
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/actions/CryptoExchangeActions.tsx]: removed

File: src/actions/ExchangeInfoActions.ts [hunk 4]
```diff
@@ -13,12 +13,15 @@ const asDisableAsset = asObject({
...
+const asDisablePluginsMap = asMaybe(asObject(asBoolean), {})
+
 export const asExchangeInfo = asObject({
   swap: asObject({
     disableAssets: asObject({
       source: asArray(asDisableAsset),
       destination: asArray(asDisableAsset)
-    })
+    }),
+    disablePlugins: asMaybe(asDisablePluginsMap, {})
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/actions/ExchangeInfoActions.ts]: removed

File: src/components/scenes/GuiPluginListScene.tsx (hunk 1 above)
Comment (peachbits) [reply src/components/scenes/GuiPluginListScene.tsx]: Replaced cleaner to give more control over pluginIds in list view and within plugins like creditcard

File: src/plugins/gui/fiatPluginTypes.ts (hunk 2 above)
Comment (peachbits) [reply src/plugins/gui/fiatPluginTypes.ts:63]: done

File: src/components/scenes/GuiPluginListScene.tsx [hunk 5]
```diff
@@ -343,7 +343,7 @@ class GuiPluginList extends React.PureComponent<Props, State> {
     // Filter disabled plugins:
     const activePlugins = bestOfPlugins(accountPlugins, accountReferral, undefined)
     plugins = plugins.filter(plugin => !activePlugins.disabled[plugin.pluginId])
-    plugins = plugins.filter(plugin => !disablePlugins[plugin.pluginId])
+    plugins = plugins.filter(plugin => disablePlugins[plugin.pluginId] !== true)
```
Comment (paullinator) [inline src/components/scenes/GuiPluginListScene.tsx:350]: This didn't consider my comment that you need to change this to `[plugin.id]`.

`[plugin.pluginId]` is shared across multiple payment methods so it doesn't allow us to disable with fine granularity.

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/components/scenes/GuiPluginListScene.tsx [hunk 6]
```diff
@@ -260,7 +260,11 @@ class GuiPluginList extends React.PureComponent<Props, State> {
 
     const regionCode = { countryCode }
     if (plugin.nativePlugin != null) {
-      await executePlugin({ disablePlugins, guiPlugin: plugin, regionCode, paymentType, navigation, account })
+      const filteredDisablePlugins: { [pluginId: string]: true } = {}
+      for (const [key, value] of Object.entries(disablePlugins[plugin.pluginId] ?? {})) {
```
Comment (paullinator) [inline src/components/scenes/GuiPluginListScene.tsx:264]: also change to `[plugin.id]`

**Review by paullinator — CHANGES_REQUESTED** [review]

**Review by paullinator — APPROVED** [review]

**Pull request merged**